File: 34 - Employee List Removal/Python/main.py
def print_list(l):
    for i in l:
        print(i)


def main():
    employees = [
        "alice",
        "bob",
        "charlie",
        "dave",
        "eve",
    ]
    print(f"There are {len(employees)} employees:")
    print_list(employees)
    name_to_rm = input("\nEnter an employee name to remove: ")
    if name_to_rm not in employees:
        print("That employee does not exist.")
    else:
        employees.remove(name_to_rm)
        print(f"\nThere are {len(employees)} employees:")
        print_list(employees)


if __name__ == "__main__":
    main()

File: 34 - Employee List Removal/Golang/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func printSList(list []string) {
	for _, v := range list {
		fmt.Println(v)
	}
}

func removeFromSList(list []string, s string) []string {
	var newList []string
	for _, val := range list {
		if val != s {
			newList = append(newList, val)
		}
	}
	return newList
}

func stdinput(output string) string {
	scanner := bufio.NewScanner(os.Stdin)
    fmt.Print(output)
    scanner.Scan()
    return scanner.Text()
}

func main() {
	employees := []string{
		"alice",
		"bob",
		"charlie",
		"dave",
		"eve",
	}
	fmt.Printf("There are %d employees\n", len(employees))
	printSList(employees)
    nameToRemove := stdinput("What is the name of the employee to remove? ")
    employees = removeFromSList(employees, nameToRemove)
	fmt.Printf("There are %d employees\n", len(employees))
    printSList(employees)
}
